apps/user_ldap/lib/Mapping/AbstractMapping.php

```diff
@@ -123,8 +123,7 @@ protected function modify($query, $parameters) {
 	 */
 	public function getDNByName($name) {
 		$dn = array_search($name, $this->cache);
-		if ($dn === false) {
-			$dn = $this->getXbyY('ldap_dn', 'owncloud_name', $name);
+		if ($dn === false && $dn = $this->getXbyY('ldap_dn', 'owncloud_name', $name)) {
```

Comment:
IMO this looks like an accidental assignment. How about

```suggestion
		if ($dn === false && ($dn = $this->getXbyY('ldap_dn', 'owncloud_name', $name)) !== false) {
```

Reply:
it's more verbose. The method returns either string or false. But of course it makes sense when otherwise it looks confusing :)